File: ciphers/SubsitutionCipher.java
package subsitution_cipher;

import java.util.HashMap;
import java.util.Map;
import java.util.Random;
import java.util.Scanner;

public class SubsitutionCipher {
    
    private static final Map<Character, Character> encryptMap = new HashMap<>();
    private static final Map<Character, Character> decryptMap = new HashMap<>();
    private static  Random random = new Random();

    static{
        for (int i = 32; i < 126; i++){
            int randomNum = random.nextInt(95) + 32;
            boolean done = false;
            while(!(done)){
                if (!(decryptMap.containsKey(((char)randomNum)))){
                    encryptMap.put((char)i, (char)randomNum);
                    decryptMap.put((char)randomNum, (char)i);
                    done = true;
                }
                else
                    randomNum = random.nextInt(95) + 32;
            }
        }
    }

    public String encrpyt(String plainText){
        String s = "";
        for (int i = 0; i < plainText.length(); i++)
            s += encryptMap.get(plainText.charAt(i));
        return s;
    }

    public String decrypt(String encryptedText){
        String s = "";
        for (int i = 0; i < encryptedText.length(); i++)
            s += decryptMap.get(encryptedText.charAt(i));
        return s;
    }

    public static void main(String[] args) {
        SubsitutionCipher cipher = new SubsitutionCipher();
        Scanner scanner = new Scanner(System.in);
        boolean done = false;
        while (!(done)){
            System.out.print("What is the text you would like to encrypt?: ");
            String plainText = scanner.nextLine();
            String encryptedText = cipher.encrpyt(plainText.strip());
            System.out.println("Here is your encrypted text: " + encryptedText);
            System.out.print("Would you like to decrypt it? y/n/exit: ");
            String yesornoDecrypt = scanner.nextLine();
            switch(yesornoDecrypt.strip().toLowerCase()){
                case "y":
                    System.out.println(cipher.decrypt(encryptedText));
                case "n":
                    break;
                case "exit":
                    done = true;
                    scanner.close();
                    break;
                default:
                    System.out.println("Not a valid command");
            }
        }

    }
}

File: ciphers/subsitution_cipher.py
import random

ENCRYPT_MAP = {}
DECRYPT_MAP = {}

rand = random.Random()
for i in range(58, 127):
    randomNum = rand.randint(58, 126)
    done = False
    while not done:
        if chr(randomNum) not in DECRYPT_MAP.keys():
            ENCRYPT_MAP[chr(i)] = chr(randomNum)
            DECRYPT_MAP[chr(randomNum)] = chr(i)
            done = True
        else:
            randomNum = rand.randint(58, 126)

def encrypt(plaintext):
    s = ""
    for i in range(len(plaintext)):
        s += ENCRYPT_MAP.get(plaintext[i])
    return s

def decrypt(encrypted_text):
    s = ""
    for i in range(len(encrypted_text)):
        s += DECRYPT_MAP.get(encrypted_text[i])
    return s

def main():
    done = False
    while not done:
        plaintext = input("What would you like to encrypt?: ")
        encrypted_text = encrypt(plaintext)
        print(encrypted_text)
        yesornoDecrypt = input("Would you like to ao decrypt your text or exit? y/n/exit: ")
        if yesornoDecrypt.strip().lower() == 'y':
            print(decrypt(encrypted_text))
        elif yesornoDecrypt.strip().lower() == 'n':
            break
        elif yesornoDecrypt.strip().lower() == "exit":
            done = True
        else:
            print("Not a valid command")

if __name__ == "__main__":
    main()
